org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateServiceProviderCatalogSingleton.mtl

```diff
@@ -216,60 +192,43 @@ public class [javaClassNameForSingleton(aServiceProviderCatalog) /]
 
     public static ServiceProvider create[aServiceProvider.javaName(true)/](final [javaClassName(aServiceProvider) /] serviceProviderInfo) 
             throws OslcCoreApplicationException, URISyntaxException, IllegalArgumentException {
-        String identifier = [aServiceProvider.javaName(false)/]Identifier([for (instanceCompositeID: String | instanceCompositeID(aServiceProvider)) separator(', ')]serviceProviderInfo.[instanceCompositeID /][/for]);
-        if (contains[aServiceProvider.javaName(true)/]([for (instanceCompositeID: String | instanceCompositeID(aServiceProvider)) separator(', ')]serviceProviderInfo.[instanceCompositeID /][/for])) {
+        String identifier = [javaClassNameForFactory(aServiceProvider)/].constructIdentifier(serviceProviderInfo);
+        if (contains[aServiceProvider.javaName(true)/](serviceProviderInfo)) {
```

Comment:
this check should be in the registration not in creation

Reply:
Sure. This whole create method should then be removed from the Singleton class. Just call the Create methods from the SPFactory. Will fix.

Reply:
no, I mean that the check should be done in the register method. It is still in the SPC and still needs to call the contains method. but yes, create methods can be removed from the factory.

Reply:
I think I understood. Fixed in latest patc.